--- AudioEngine.h	2013-07-09 17:45:44.000000000 +0000
+++ AudioEngine_patched.h	2013-07-09 18:55:47.952910680 +0000
@@ -24,6 +24,7 @@
 #include "../Typedefs.h"
 
 #include <QStringList>
+#include <phonon/Path>
 
 #include "DllMacro.h"
 
@@ -62,6 +63,8 @@
     qint64 currentTime() const;
     qint64 currentTrackTotalTime() const;
 
+    Phonon::Path* getAudioPath() { return &this->audioPath; }
+
 public slots:
     void playPause();
     void play();
@@ -148,6 +151,7 @@
 
     Q_DECLARE_PRIVATE( AudioEngine );
     AudioEnginePrivate* d_ptr;
+    Phonon::Path audioPath;
 };
 
 #endif // AUDIOENGINE_H
--- AudioEngine.cpp	2013-07-09 17:45:44.000000000 +0000
+++ AudioEngine_patched.cpp	2013-07-09 18:50:32.284901987 +0000
@@ -191,7 +191,7 @@
 
     d->mediaObject = new Phonon::MediaObject( this );
     d->audioOutput = new Phonon::AudioOutput( Phonon::MusicCategory, this );
-    Phonon::createPath( d->mediaObject, d->audioOutput );
+    this->audioPath = Phonon::createPath( d->mediaObject, d->audioOutput );
 
     d->mediaObject->setTickInterval( 150 );
     connect( d->mediaObject, SIGNAL( stateChanged( Phonon::State, Phonon::State ) ), d_func(), SLOT( onStateChanged( Phonon::State, Phonon::State ) ) );
